### ISSUE TITLE
Register two instance of type for one type in controller from external dll.

### ISSUE THREAD

**Issue by Nelo-cool** [seesharper/LightInject#208]
Hello:) Please, Help.
Begin conditions:
1. Self hosted web api
2. Host = console appliaction
3. Framework 4.5
4. LightInject last version from nuget
Goal:
Load controller from external dll and using DI container
Result:
1. Impl custom resolver 
CustomAssemblyResolver : DefaultAssembliesResolver
{
        public override ICollection<Assembly> GetAssemblies()
        {
            var controllersAssembly = Assembly.LoadFrom(@"Controllers.dll");
            var baseAssemblies = base.GetAssemblies();
            baseAssemblies.Add(controllersAssembly);

            return baseAssemblies;
        }
}

2. Replace resolver in Program.cs

var config = new HttpSelfHostConfiguration("http://localhost:5555");
config.Services.Replace(typeof(IAssembliesResolver), new CustomAssemblyResolver());

3.Call function for register IOC after rows from step 2

 IoCConfig.RegisterIoC(config);

In this function (include class):

public static class IoCConfig
{
        private readonly static ServiceContainer _container = new ServiceContainer();

        static IoCConfig()
        {
            _container.Register<ITestService>(x => new TestService(), new PerContainerLifetime());
        }

        public static void RegisterIoC(HttpSelfHostConfiguration configuration)
        {   
            _container.RegisterApiControllers();
            _container.EnableWebApi(configuration);
        }
 }

4. Map attribute from TestController after rows form step 3

config.MapHttpAttributeRoutes();

5. Run server

using (var server = new HttpSelfHostServer(config))
{
       server.OpenAsync().Wait();
       Console.ReadLine();
}

6. TestController implementation from Connrtolles.dll:
public class TestController : ApiController
{
        private IDPMService _service = null;
        
       public TestController(IDPMService service)
       {
            _service = service;
       }

       [Route("Test")]
       public List<Tuple<Guid, string>> Get()
       {
            return _service.GetService();
       }
  }

7. Run programm and browser (IE, Opera...) and write "http://localhost:5555/Test" push load and i see:
<Error>
<Message>
No HTTP resource was found that matches the request URI 'http://localhost:5555/Test'.
</Message>
<MessageDetail>No route data was found for this request.</MessageDetail>
</Error>

8.1 If disable (comment) IoC from step 3 this solution work fine. But not DI:(
8.2. If add after step 4 rows: 
config.Routes.MapHttpRoute("Test1", "Test", new { controller = "Test", action = "Get", });
I see:

<Error>
<Message>An error has occurred.</Message>
<ExceptionMessage>
Multiple types were found that match the controller named 'Test'. This can happen if the route that services this request ('Test') found multiple controllers defined with the same name but differing namespaces, which is not supported. The request for 'Test' has found the following matching controllers: Controllers.TestController
</ExceptionMessage>
<ExceptionType>System.InvalidOperationException</ExceptionType>
<StackTrace>
в System.Web.Http.Dispatcher.DefaultHttpControllerSelector.SelectController(HttpRequestMessage request) в Host.CustomHttpControllerSelector.SelectController(HttpRequestMessage request) в c:\Users\user\Documents\Visual Studio 2013\Projects\ConsoleApplication4\ConsoleApplication4\CustomHttpControllerSelector.cs:строка 23 в System.Web.Http.Dispatcher.HttpControllerDispatcher.<SendAsync>d__1.MoveNext()
</StackTrace>
</Error>

Solutions? This is a bug? Or?

**Comment by seesharper** [comment]
Could you upload the complete example somewhere or even better just make it a repository here on github so that I can just run it ?

**Comment by Nelo-cool** [comment]
Yes, I can tomorrow, just my example at work. Thanks for your answer!

**Comment by Nelo-cool** [comment]
Hello! Download at https://thisislogic.ru/ExampleLightInject.zip
